pinot-common/src/main/java/org/apache/pinot/common/restlet/resources/SegmentsReloadCheckResponse.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.restlet.resources;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+
+/**
+ * This class gives the data of a server if there exists any segments that need to be reloaded
+ *
+ * It has details of server id and returns true/false if there are any segments to be reloaded or not.
+ */
+public class SegmentsReloadCheckResponse {
+  private final boolean _needReload;
+  private final String _serverInstanceId;
+
+  public String getServerInstanceId() {
+    return _serverInstanceId;
+  }
+
+  public boolean getReload() {
```

Comment:
This method name might not work for JSON serialization. I think it will be mistakenly serialized as `reload`. Can you add a test to verify? We might need to rename it to `isNeedReload()`, or explicitly annotate it with `@JsonProperty("needReload")`. If we explicitly annotate, a better name could be `isReloadNeeded()`

Reply:
We have it, it's just little complex to separate out server instance when doing multi get request which is why just returning it well in the response to make it simpler. I will look into this later. For now, we can include instanceId as well I think